### PR TITLE
use alpaka in PMacc buffer types

### PR DESCRIPTION
**Pull request by psychocoderHPC**
- add alpaka buffer
- add new interface method `getMemBufView()` to buffer

~~This pull request needs #4 but we not compile anything therefore no rebase is needed.~~

### CODE REVIEW

File: src/libPMacc/include/memory/buffers/Buffer.hpp [hunk 1]
```diff
@@ -55,9 +69,19 @@ namespace PMacc
...
+            >(
+                Environment<>::get().DeviceManager().getHostDevice(),
+                static_cast<alpaka::MemSize>(1u)
+            )
+        )
         {
-            CUDA_CHECK(cudaMallocHost(&current_size, sizeof (size_t)));
+            current_size = ::alpaka::mem::view::getPtrNative(m_memBufCurrentSizeHost);
             *current_size = size.productOfComponents();
         }
```
Comment (erikzenker) [inline src/libPMacc/include/memory/buffers/Buffer.hpp]: Does current_size really needs to be a pointer ?

Comment (psychocoderHPC) [inline src/libPMacc/include/memory/buffers/Buffer.hpp]: Yes to support copy to device.

**Comment by psychocoderHPC** [comment]
rebased

**Comment by erikzenker** [comment]
Brilliant :high_brightness:

**Pull request merged**